### ISSUE TITLE
Importing custom NER tagger into flair and using .predict(). Using BIOUL tagging scheme.

### ISSUE THREAD

**Issue by raceee** [zalandoresearch/flair#787]
A clear and concise description of what you want to know:
I want to make a custom NER tagger that can be used inside of Flair to `.predict()` custom tags of words in a text I give to it. Very similar to the question brought up in #207.

In tutorial 2:
`from flair.models import SequenceTagger`
`tagger = SequenceTagger.load('ner')`

Would I make a BIOUL scheme of my text like qiuwei said in #207 and then put the file path into the `.load()` like:

`tagger = SequenceTagger.load('path_to_BIOUL')`?

When it comes to BIOUL tagging scheme how much data should I tag? More the better?

Thanks!

**Comment by henry32144** [comment]
I was using add_tag method to add tag in version 0.4.1, I don't know whether it has been changed in more recent version.

This is a small example.
```
from flair.data import TaggedCorpus
from flair.data import Sentence
from flair.data_fetcher import  NLPTaskDataFetcher, NLPTask
from flair.embeddings import TokenEmbeddings, WordEmbeddings, StackedEmbeddings, CharLMEmbeddings, CharacterEmbeddings
from segtok.segmenter import split_single
from typing import List
import torch

# Convert string to sentence, this will auto s
test_sentence = Sentence("Egg fried rice is delicious .")
tags = ["B-Food", "I-Food","L-Food", "O", "O", "O"]

for x in range(len(test_sentence)):
    test_sentence[x].add_tag('ner', tags[x])

test_sentence.to_tagged_string()
# Output
# 'Egg <B-Food> fried <I-Food> rice <L-Food> is delicious .'

# Let's assume you have already create lists which contain tagged sentences 
corpus = TaggedCorpus(sentences, testa, testb)

# Make the tag dictionary from the corpus
tag_dictionary = corpus.make_tag_dictionary(tag_type="ner")
print(tag_dictionary.idx2item)

stacked_embeddings = StackedEmbeddings([
                                        WordEmbeddings('glove'), 
                                        FlairEmbeddings('news-forward'), 
                                        FlairEmbeddings('news-backward'),
                                       ])

# Initialize sequence tagger
from flair.models import SequenceTagger

tagger: SequenceTagger = SequenceTagger(hidden_size=256,
                                        embeddings=stacked_embeddings,
                                        tag_dictionary=tag_dictionary,
                                        tag_type="ner",
                                        use_crf=True)

# initialize trainer
from flair.trainers import ModelTrainer

trainer: ModelTrainer = ModelTrainer(tagger, corpus)

# Start training
trainer.train('model',
              learning_rate=0.1,
              mini_batch_size=32,
              max_epochs=80)
```

**Comment by alanakbik** [comment]
Hello @raceee - yes like @henry32144 wrote you need to create your own corpus and then use the `ModelTrainer` to train your own model. 

You can follow [these instructions](https://github.com/zalandoresearch/flair/blob/master/resources/docs/TUTORIAL_6_CORPUS.md#reading-your-own-sequence-labeling-dataset) to create your own sequence labeling dataset and read it into a `Corpus` object. 

You can then use this `Corpus` to train your own model by following [these instructions](https://github.com/zalandoresearch/flair/blob/master/resources/docs/TUTORIAL_7_TRAINING_A_MODEL.md#training-a-sequence-labeling-model). Just replace the example corpus in the code snippet here with your own `Corpus` that you produced with the other instructions. That gives you a model that you can load into the `SequenceTagger` to make your own predictions. Hope this helps!

**Comment by DecentMakeover** [comment]
Hi 

Not able to import TaggedCorpus !! Any idea why?

Thanks

**Comment by alanakbik** [comment]
As of Flair 0.4.2, the `TaggedCorpus` object no longer exists. It's now simply `Corpus`. You can follow [these instructions](https://github.com/zalandoresearch/flair/blob/master/resources/docs/TUTORIAL_6_CORPUS.md#reading-your-own-sequence-labeling-dataset) to create your own sequence labeling dataset and read it into a Corpus object.

**Comment by DecentMakeover** [comment]
Ah , 

Thanks a lot for the quick response.Ill look into it.

**Comment by m-michalek** [comment]
Hello guys!

I followed the link and constructed my own training data.
Since I am only interested in a NER model and have no information about the PoS tags of my data it looks like this:

George PER
Washington PER
went O
to O
Washington LOC

But the training always stops at the first epoch, sometimes it is able to get to iteration 6 but sometimes it stops before iteration 1. I am wondering what I am doing wrong because when I use the WIKINER_ENGLISH data the training works fine.

![Capture](https://user-images.githubusercontent.com/20742170/60503860-cee0be80-9cc0-11e9-903a-dc0669111e3a.JPG)

From the image, you can see my dataset is really small, flair is able to read it and I am getting a loss but training gets killed in epoch 1 iteration 6. I have tested my code on multiple machines with the same result. This is the code I used:

Below is my code I used

I thank you for any help

Code:

columns = {0: 'text', 1: 'ner'}

data_folder = './'

corpus: Corpus = ColumnCorpus(data_folder, columns,
                              train_file='train.txt',
                              test_file='test.txt',
                              dev_file='dev.txt')
print(corpus)

tag_type = 'ner'

tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
print(tag_dictionary.idx2item)

embedding_types: List[TokenEmbeddings] = [
    WordEmbeddings('glove'),
]

embeddings: StackedEmbeddings = StackedEmbeddings(embeddings=embedding_types)

from flair.models import SequenceTagger

tagger: SequenceTagger = SequenceTagger(hidden_size=256,
                                        embeddings=embeddings,
                                        tag_dictionary=tag_dictionary,
                                        tag_type=tag_type,
                                        use_crf=True)

from flair.trainers import ModelTrainer

trainer: ModelTrainer = ModelTrainer(tagger, corpus)

trainer.train('resources/taggers/ner_custom_model',
              learning_rate=0.1,
              mini_batch_size=32,
              max_epochs=150)

from flair.visual.training_curves import Plotter
plotter = Plotter()
plotter.plot_training_curves('resources/taggers/ner_custom_model/loss.tsv')
plotter.plot_weights('resources/taggers/ner_custom_model/weights.txt')
